From 7be750cf75cf428f7de2a891d6e40db31eab4736 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Fri, 11 Oct 2013 15:27:49 -0700
Subject: [PATCH 276/518] MIPS: OCTEON: Allow CONFIG_CAVIUM_CN63XXP1 to be
 disabled.

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: 3793d0395f913a71efb29545611bfa9156aceb82
Description:

It is advantageous to use CONFIG_CAVIUM_OCTEON2 with OCTEON III CPU
cores as they have the same instruction set.  However, OCTEON III
doesn't require the CN63XXP1 workaround, so make it optional.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/cavium-octeon/Kconfig | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/mips/cavium-octeon/Kconfig b/arch/mips/cavium-octeon/Kconfig
index 5177edb..5da5486 100644
--- a/arch/mips/cavium-octeon/Kconfig
+++ b/arch/mips/cavium-octeon/Kconfig
@@ -1,8 +1,7 @@
 if CPU_CAVIUM_OCTEON
 
 config CAVIUM_OCTEON2
-	bool "Build the kernel to be used only on OCTEON II processor cores"
-	select CAVIUM_CN63XXP1
+	bool "Build the kernel to be used only on OCTEON II or OCTEON III processor cores"
 	default "n"
 	help
 	  This option enables the generation of Octeon2 specific
-- 
1.9.1

